File: greedy.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
input:
    printf("O hai! How much change is owed?\n");
    double dollar = get_double();
    if (dollar < 0)
    {
        goto input;
    }
    else
    {
        int penny = dollar * 100;
        if (penny >= 25)
        {
            int quarters = (penny / 25);
            int qremainder = (penny % 25);
            int dimes = (qremainder / 10);
            int dremainder = (qremainder % 10);
            int nickels = (dremainder / 5);
            int p = (dremainder % 5);
            printf("%i\n", quarters + dimes + nickels + p);
        }
        else if (penny >= 10)
        {
            int dimes = (penny / 10);
            int dremainder = (penny % 10);
            int nickels = (dremainder / 5);
            int p = (dremainder % 5);
            printf("%i\n", dimes + nickels + p);
        }
        else if (penny >= 5)
        {
            int nickels = (penny / 5);
            int p = (penny % 5);
            printf("%i\n", nickels + p);
        }
        else
        {
            printf("%i\n", penny);
        }
    }
}

File: greedy.py
import cs50

dollar = cs50.get_float("O hai! How much change is owed?")

while (dollar < 0):
    dollar = cs50.get_float("How much change is owed?")

penny = int(dollar * 100)

if penny >= 25:
    quarters = penny // 25
    qremainder = penny % 25
    dimes = qremainder // 10
    dremainder = qremainder % 10
    nickels = dremainder // 5
    p = dremainder % 5
    print(quarters + dimes + nickels + p)
elif penny >= 10:
    dimes = penny // 10
    dremainder = penny % 10
    nickels = dremainder // 5
    p = dremainder % 5
    print(dimes + nickels + p)
elif penny >= 5:
    nickels = penny // 5
    p = penny % 5
    print(nickels + p)
else:
    print(penny)